### ISSUE TITLE
Issue with Image Extractor and waveforms shape

### ISSUE THREAD

**Issue by vuillaut** [cta-observatory/cta-lstchain#482]
Related to #475 

I finally could reproduce the error, so I am reporting with minimal code:

```
from ctapipe.io import event_source
from ctapipe.utils import get_dataset_path
from ctapipe.image.extractor import LocalPeakWindowSum, GlobalPeakWindowSum, NeighborPeakWindowSum

local_ex = LocalPeakWindowSum(subarray=reader.subarray)
global_ex = GlobalPeakWindowSum(subarray=reader.subarray)
neighbor_ex = NeighborPeakWindowSum(subarray=reader.subarray)

reader = event_source(get_dataset_path('gamma_test.simtel.gz'))

selected_gain_channel = np.zeros((2,1855), dtype=np.int)
waveform=np.ones((2,1855,40))
waveform[0]=3
```

The `LocalPeak` works fine with 2 channels:

```
local_ex(waveform, 1, selected_gain_channel)
Out[6]:
(array([[12.760198 , 12.760198 , 12.760198 , ..., 12.760198 , 12.760198 ,
         12.760198 ],
        [ 4.2533994,  4.2533994,  4.2533994, ...,  4.2533994,  4.2533994,
          4.2533994]], dtype=float32),
 array([[1.5, 1.5, 1.5, ..., 1.5, 1.5, 1.5],
        [1.5, 1.5, 1.5, ..., 1.5, 1.5, 1.5]], dtype=float32))
```

The `GlobalPeak` returns an error concerning the shape:

```
global_ex(waveform, 1, selected_gain_channel)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-6db7b3ed2c82> in <module>
----> 1 global_ex(waveform, 1, selected_gain_channel)

~/opt/anaconda3/envs/py3.8/lib/python3.7/site-packages/ctapipe/image/extractor.py in __call__(self, waveforms, telid, selected_gain_channel)
    446             self.window_width.tel[telid],
    447             self.window_shift.tel[telid],
--> 448             self.sampling_rate[telid],
    449         )
    450         charge *= self._calculate_correction(telid=telid)[selected_gain_channel]

ValueError: operands could not be broadcast together with remapped shapes [original->remapped]: (2,1855,40)->(2,1855) (2,)->(2) ()->() ()->() ()->() and requested shape ()
```
The `NeighborPeak` makes a segmentation fault (most probably due to numba and wrong waveform shape):

```
neighbor_ex(waveform, 1, selected_gain_channel)
[1]    57469 segmentation fault  ipython
```

**Comment by vuillaut** [comment]
The docstring of `extract_around_peak` is clear about the expected shape:
```
    waveforms : ndarray
        Waveforms stored in a numpy array.
        Shape: (n_pix, n_samples)
```
`LocalPeak` works only by chance.

**Comment by MaxNoe** [comment]
Yes, also looking at the changelog of ctapipe 0.8, it is pretty clear that it expects gain selection to have already happened.

**Comment by vuillaut** [comment]
> Yes, also looking at the changelog of ctapipe 0.8, it is pretty clear that it expects gain selection to have already happened.

Yes, I got very confused when retesting with the `LocalPeak` and noticed it was working, I thought we could get away with multiple channels.

**Comment by vuillaut** [comment]
Solved in #475
